plugin-testlib/src/test/kotlin/io/spine/tools/gragle/testing/ReplacementTest.kt

```diff
@@ -0,0 +1,172 @@
+
+    @BeforeEach
+    fun setUp(@TempDir tempDir: Path) {
+        folder = tempDir.toFile()
+    }
+
+    @Test
+    fun `frame the token name with '@' symbols`() {
+        val original = "mytoken"
+        val actual = Replacement(original, "").token()
+        assertThat(actual)
+            .isEqualTo("@$original@")
+    }
+
+    @Test
+    fun `not allow empty tokens`() {
+        assertThrows<IllegalArgumentException> { Replacement("", "value")  }
+    }
+
+    @Test
+    fun `allow empty values-to-replace-with`() {
+        assertThat(Replacement("sometoken", "").value)
+            .isEmpty()
+    }
+
+    @Nested
+    inner class `replace all token occurrences` {
+
+        @Test
+        fun `in a file`() {
+            val ref = TestData(folder, TEXT)
+            val value = "replaced"
+            Replacement(TOKEN_NAME, value).replaceIn(ref.file)
+            ref.assertReplaced(TOKEN_NAME, value)
+        }
+
+        @Test
+        fun `in files residing in a folder and its subfolders`() {
+            val refs = generateFiles(folder)
+
+            val value = "recursively-replaced"
+            Replacement(TOKEN_NAME, value)
+                .replaceIn(folder)
+            for (ref in refs) {
+                ref.assertReplaced(TOKEN_NAME, value)
+            }
+        }
+
+        @Test
+        fun `in all files in the folder recursively, excluding some folder`() {
+            val toReplace = generateFiles(folder)
+
+            val excludedFolder = folder.resolve("untouchable")
+            excludedFolder.mkdirs()
+            val toExclude = generateFiles(excludedFolder)
+
+            val value = "recursively-replaced"
+            Replacement(TOKEN_NAME, value)
+                .replaceIn(folder, excludedFolder)
+
+            for (ref in toReplace) {
+                ref.assertReplaced(TOKEN_NAME, value)
+            }
+            for (ref in toExclude) {
+                ref.assertNotReplaced()
+            }
+        }
+    }
+
+    private fun generateFiles(folder: File): ArrayList<TestData> {
+        val refs = ArrayList<TestData>()
+        for (i in 0..10) {
+            refs.add(TestData(folder, TEXT))
+        }
+        return refs
+    }
+
+    companion object {
+
+        private val random: Random = SecureRandom()
+
+        const val TOKEN_NAME = "TEST_TOKEN"
+        const val TEXT = "This is a `@$TOKEN_NAME@` which should be replaced with " +
+                "`@$TOKEN_NAME@`. \n\n\n And this `@TEST@ should remain the same."
+
+        /**
+         * Creates a test file with the specified [content] inside the passed [parentFolder].
+         *
+         * Prior to creating the file, establishes some number of sub-folders, to which the file
+         * is put. The number lies between zero and two.
+         *
+         * The name of the file and the names of folders are based on UUIDs.
+         */
+        class TestData(private val parentFolder: File, val content: String) {
```

Comment:
Why is this class nested in a `companion object`? It could be placed directly inside the test suite class, this wouldn't, unlike in Java, bind it to the host class.